### PR TITLE
Make sidebars responsive

### LINKED ISSUE #8133: Settings page on mobile layout
It would be nice to have a responsive settings page. Probably with some sort of drop down menu instead of the side menu on mobile. I'm willing to help with the styling, but I need some suggestions to get it the way you'd like.

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

Step 1 to making sidebars responsive. Shifts the sidebar to a row that scrolls on smaller screens. Also made the `<Heading1/>` slightly smaller on small screens.

| Header | Header |
|--------|--------|
| image | image | 

**Follow Up Steps**
* Explore using a dropdown for the sidebar navigation items versus scrollable menu.
* Explore adding a scrolling indicator on left/right edge of menu when there's more options off-screen.
* Currently navigation on the submenu results in a full route re-render, which resets the scroll position of the menu. This PR shifts the active item into visibility, but it isn't able to maintain scroll position due to the re-render. We can address this in a followup by using nested routes w/ react-router, and only re-render the portions of the page that change as much as possible.

## Related Issue(s)

Fixes #8133

## How to test

**Preview Env:** https://bmh-respon4a5450a2ee.preview.gitpod-dev.com/user/account
* Visit some of the pages with sidebar nav (user/org settings) on mobile and desktop.

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@selfcontained: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by gtsiolis** [comment]
Looking at this now! 👀 

[Install Gitpod](https://github.com/gitpod-io/gitpod/actions/runs/4506169616/jobs/7938827701?pr=17010) job seems successful.

**Review by geropl — APPROVED** [review]
Code LGTM, tested and works nicely :art: 

/hold For @gtsiolis 's feedback

File: components/dashboard/src/components/PageWithSubMenu.tsx
Comment (gtsiolis) [inline components/dashboard/src/components/PageWithSubMenu.tsx:72]: **suggestion**: Probably not the best line to comment on this, but the initial idea waw to reuse the same tabs pattern we use in the dashboard. It should be ok to leave is as is for this PR but it'd be nicer for users to face a pattern they've seen before. Thoughts?

| BEFORE | AFTER |
|-|-|
| bmh-respon4a5450a2ee preview gitpod-dev com_usage(Pixel 5) | bmh-respon4a5450a2ee preview gitpod-dev com_usage(Pixel 5) (1) |

Comment (gtsiolis) [inline components/dashboard/src/components/PageWithSubMenu.tsx:72]: **thought**: Going with the tabs pattern will also help decrease the gap between the menu items and providing a hint for menu items overflowing the viewport.

File: components/dashboard/src/components/PageWithSubMenu.tsx
Comment (gtsiolis) [inline components/dashboard/src/components/PageWithSubMenu.tsx:36]: **question**: What do you think of making the horizontal scrollbar invisible? This works ok now with the pill-like design, but could clash with the tabs.

Screenshot 2023-03-24 at 13 06 10

File: components/dashboard/src/components/PageWithSubMenu.tsx
Comment (gtsiolis) [inline components/dashboard/src/components/PageWithSubMenu.tsx:84]: **suggestion**: Probably not the best line to comment about this, but could we not shift the menu items when an item is clicked? Noticed that clicking on an item moves it to the left side of the scrollable area, which sometimes seems a bit jarring, no?


https://user-images.githubusercontent.com/120486/227506186-68bcb395-dc8b-470b-af7e-679f09f2e797.mov

**Review by gtsiolis — APPROVED** [review]
Thanks for making this change, @selfcontained!

Left a few comments below following our discussion around this menu.

Approving additionally to unblock this bu holding in case you'd like to address the UX comments below.

/hold

File: components/dashboard/src/components/PageWithSubMenu.tsx
Comment (selfcontained) [reply components/dashboard/src/components/PageWithSubMenu.tsx:84]: Yah, I didn't like this either, tried to explain it a bit in the PR description, but it's buried in there:

> Currently navigation on the submenu results in a full route re-render, which resets the scroll position of the menu. This PR shifts the active item into visibility, but it isn't able to maintain scroll position due to the re-render. We can address this in a followup by using nested routes w/ react-router, and only re-render the portions of the page that change as much as possible.

Another thing to consider is if we did a dropdown for nav instead of this scrollbar or a tab-bar, it would avoid that issue.

**Pull request merged**
